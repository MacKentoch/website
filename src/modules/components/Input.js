import glamorous from 'glamorous'

const Input = glamorous.input(
  {
    borderRadius: 3,
    display: 'inline-block',
    height: 38,
    padding: '0 10px',
    fontSize: 15,
    lineHeight: '38px',
    transition: 'border-color 200ms',
    borderWidth: '1px',
    borderStyle: 'solid',
  },
  (props, theme) => ({
    fontFamily: theme.fontFamilies.primary,
    borderColor: theme.colors.gray,
    color: theme.colors.grayDark,
    ':focus': {
      borderColor: theme.colors.primary,
      outline: 0,
    },
  }),
)

export default Input
