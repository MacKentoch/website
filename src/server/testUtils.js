import * as database from 'server/services/database'

export const useDatabase = () => {
  let knex

  beforeAll(async () => {
    knex = database.connect()
  })

  afterAll(async () => {
    await database.disconnect()
  })

  beforeEach(async () => {
    await knex.seed.run()
  })
}
