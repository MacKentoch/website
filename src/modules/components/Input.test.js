import React from 'react'
import { shallowWithTheme } from 'test/testUtils'
import Input from './Input'

describe('Input ', () => {
  it('should render an input', () => {
    expect(shallowWithTheme(<Input />)).toMatchSnapshotWithGlamor()
  })
})
