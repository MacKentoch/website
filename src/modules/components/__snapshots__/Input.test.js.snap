// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input  should render an input 1`] = `
.css-1camfti,
[data-css-1camfti] {
  -webkit-transition: border-color 200ms;
  border-radius: 3px;
  height: 38px;
  padding: 0 10px;
  font-size: 15px;
  line-height: 38px;
  display: inline-block;
  border-width: 1px;
  border-style: solid;
  font-family: 'Roboto', sans-serif;
  border-color: #CCC;
  color: #333;
  transition: border-color 200ms;
}

.css-1camfti:focus,
[data-css-1camfti]:focus {
  border-color: #333;
  outline: 0;
}

<input
  className="css-1camfti"
/>
`;
